app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperOnboarding.kt

```diff
@@ -0,0 +1,141 @@
+ * @param onCloseClicked Callback for when the close button is clicked.
+ * @param onExploreMoreButtonClicked Callback for when the bottom text button is clicked.
+ * @param onSelectWallpaper Callback for when a new wallpaper is selected.
+ */
+
+@Suppress("LongParameterList")
+@ExperimentalMaterialApi
+@Composable
+fun WallpaperOnboarding(
+    wallpapers: List<Wallpaper>,
+    currentWallpaper: Wallpaper,
+    loadWallpaperResource: suspend (Wallpaper) -> Bitmap?,
+    onCloseClicked: () -> Unit,
+    onExploreMoreButtonClicked: () -> Unit,
+    onSelectWallpaper: (Wallpaper) -> Unit,
+) {
+    Surface(
+        color = FirefoxTheme.colors.layer2,
+        shape = RoundedCornerShape(topStart = 16.dp, topEnd = 16.dp),
+    ) {
+        Column(
+            modifier = Modifier.padding(16.dp),
+            horizontalAlignment = Alignment.CenterHorizontally,
+        ) {
+            Icon(
+                painter = painterResource(id = R.drawable.mozac_ic_close),
+                contentDescription = stringResource(id = R.string.close_tab),
+                tint = FirefoxTheme.colors.iconPrimary,
+                modifier = Modifier
+                    .clickable { onCloseClicked() }
+                    .size(24.dp)
+                    .align(Alignment.End),
+            )
+
+            Spacer(modifier = Modifier.height(8.dp))
+
+            Text(
+                text = stringResource(R.string.wallpapers_onboarding_dialog_title_text),
+                style = FirefoxTheme.typography.headline7,
+                color = FirefoxTheme.colors.textPrimary,
+                overflow = TextOverflow.Ellipsis,
+                maxLines = 1,
+            )
+
+            Spacer(modifier = Modifier.height(4.dp))
+
+            Text(
+                text = stringResource(R.string.wallpapers_onboarding_dialog_body_text),
+                style = FirefoxTheme.typography.caption,
+                color = FirefoxTheme.colors.textSecondary,
+                overflow = TextOverflow.Ellipsis,
+                maxLines = 1,
+            )
+
+            WallpaperThumbnails(
+                wallpapers = wallpapers,
+                defaultWallpaper = Wallpaper.Default,
+                selectedWallpaper = currentWallpaper,
+                loadWallpaperResource = { loadWallpaperResource(it) },
+                onSelectWallpaper = { onSelectWallpaper(it) },
+                verticalPadding = 16,
+                horizontalPadding = 0,
+            )
+
+            TextButton(
+                modifier = Modifier
+                    .align(Alignment.CenterHorizontally)
+                    .fillMaxWidth(),
+                onClick = { onExploreMoreButtonClicked() },
+            ) {
+                Text(
+                    text = stringResource(R.string.wallpapers_onboarding_dialog_explore_more_button_text),
+                    fontWeight = FontWeight.Bold,
+                    color = FirefoxTheme.colors.textAccent,
+                    fontSize = 14.sp,
```

Comment:
This could be `FirefoxTheme.typography.button`. We shouldn't ever have to define `fontWeight` because our typography is well defined. We also typically encourage following the order of the named parameters that you would see defined by the calling function.

```
@Composable
fun Text(
    text: String,
    modifier: Modifier = Modifier,
    color: Color = Color.Unspecified,
    fontSize: TextUnit = TextUnit.Unspecified,
    fontStyle: FontStyle? = null,
    fontWeight: FontWeight? = null,
    fontFamily: FontFamily? = null,
    letterSpacing: TextUnit = TextUnit.Unspecified,
    textDecoration: TextDecoration? = null,
    textAlign: TextAlign? = null,
    lineHeight: TextUnit = TextUnit.Unspecified,
    overflow: TextOverflow = TextOverflow.Clip,
    softWrap: Boolean = true,
    maxLines: Int = Int.MAX_VALUE,
    onTextLayout: (TextLayoutResult) -> Unit = {},
    style: TextStyle = LocalTextStyle.current
)
```
So, we would typically expect `style` to be the last named parameter.